src/utils/react-query-factory.js

```diff
@@ -0,0 +1,72 @@
+// The keys that map to enterpriseQueryKeys that is used to define specific queries
+// and used to specifically invalidate queries
+export const queryKeys = {
+  // Global key to all queries
+  all: ['enterprise'],
+  enterprise: (enterpriseUuid) => [...queryKeys.all, enterpriseUuid],
+  entitlements: ['user', 'entitlements'],
+  enterpriseLearner: () => [...queryKeys.all, 'linked-enterprise-customer-users'],
+  enterpriseCourseEnrollments: (enterpriseUuid) => [...queryKeys.enterprise(enterpriseUuid), 'enrollments'],
+  enterpriseCourseMetadata: (enterpriseUuid) => [...queryKeys.enterprise(enterpriseUuid), 'course'],
+  canRedeem: (enterpriseUuid) => [...queryKeys.enterpriseCourseMetadata(enterpriseUuid), 'can-redeem'],
+  // By feature prepended by enterpriseUuid
+  enterpriseCuration: (enterpriseUuid) => [...queryKeys.enterprise(enterpriseUuid), 'content-highlights'],
+  enterpriseCurationConfiguration: (enterpriseUuid) => [...queryKeys.enterprise(enterpriseUuid), 'configuration'],
+  // All user subsidy related queries - subscriptions, policies, codes and offers
+  allEnterpriseCustomerUserSubsidies: () => [...queryKeys.all, 'user-subsidy'],
+  // Subsidy specific keys
+  subscriptions: () => [...queryKeys.allEnterpriseCustomerUserSubsidies(), 'subscriptions'],
+  policy: () => [...queryKeys.allEnterpriseCustomerUserSubsidies(), 'policy'],
+  couponCodes: () => [...queryKeys.allEnterpriseCustomerUserSubsidies(), 'coupon-codes'],
+  offers: () => [...queryKeys.allEnterpriseCustomerUserSubsidies(), 'enterprise-learner-offers'],
+  browseAndRequest: () => [...queryKeys.allEnterpriseCustomerUserSubsidies(), 'browse-and-request'],
+  // policy specific endpoints
+  redeemablePolicies: () => [...queryKeys.policy(), 'redeemable-policies'],
+  // browse and request endpoints
+  browseAndRequestConfiguration: (enterpriseUuid) => [...queryKeys.browseAndRequest(), enterpriseUuid, 'configuration'],
+};
+
+// The queryKey for useQuery hook
+export const enterpriseQueryKeys = {
```

Comment:
At a high-level, I think another possible benefit of the `@lukemorales/query-key-factory` NPM factory here is to enable a bit cleaner approach compared to this existing pattern of query keys. I keep getting tripped up by having query key definitions split across 2 different objects, one relying on the other.

Another nice benefit of the NPM package might be that it should help to infer type definitions for the queries when accessing the query key factory.